lib/improver/utilities/temporal.py

```diff
@@ -624,3 +630,50 @@ def process(self, cube_t0, cube_t1):
             interpolated_cubes.append(single_time)
 
         return interpolated_cubes
+
+
+def extract_nearest_time_point(
+        cube, dt, time_name="time", allowed_dt_difference=None):
+    """Find the nearest time point to the time point provided.
+
+    Args:
+        cube (iris.cube.Cube):
+            Cube or CubeList that will be extracted from using the supplied
+            time_point
+        dt (datetime.datetime):
+            Datetime representation of a time that will be used within the
+            extraction from the cube supplied.
+        time_name (str):
```

Comment:
I don't think this will work with forecast_period. It might be worth noting this.

Reply:
Added the raising of an exception, if a coordinate other than "time" or "forecast_reference_time" is provided.